.gitlab/kernel_version_testing/security_agent.yml

```diff
@@ -0,0 +1,235 @@
+# Security agent overrides
+upload_dependencies_security_agent_x64:
+  extends:
+    - .package_dependencies
+  rules: !reference [.on_security_agent_changes_or_manual]
+  variables:
+    ARCHIVE_NAME: dependencies-x86_64.tar.gz
+    ARCH: amd64
+    INSTANCE_TYPE: m5d.metal
+    TEST_COMPONENT: security-agent
+
+upload_dependencies_security_agent_arm64:
+  extends:
+    - .package_dependencies
+  rules: !reference [.on_security_agent_changes_or_manual]
+  variables:
+    ARCHIVE_NAME: dependencies-arm64.tar.gz
+    ARCH: arm64
+    INSTANCE_TYPE: m6gd.metal
+    TEST_COMPONENT: security-agent
+
+kernel_matrix_testing_setup_env_security_agent_arm64:
+  extends:
+    - .kernel_matrix_testing_setup_env
+  rules: !reference [.on_security_agent_changes_or_manual]
+  variables:
+    INSTANCE_TYPE: "m6gd.metal"
+    INSTANCE_TYPE_ARG: "--instance-type-arm=$INSTANCE_TYPE"
+    ARCH: arm64
+    AMI_ID_ARG: "--arm-ami-id=$KERNEL_MATRIX_TESTING_ARM_AMI_ID"
+    LibvirtSSHKey: $CI_PROJECT_DIR/libvirt_rsa-arm
+    TEST_COMPONENT: security-agent
+
+kernel_matrix_testing_setup_env_security_agent_x64:
+  extends:
+    - .kernel_matrix_testing_setup_env
+  rules: !reference [.on_security_agent_changes_or_manual]
+  variables:
+    INSTANCE_TYPE: "m5d.metal"
+    INSTANCE_TYPE_ARG: "--instance-type-x86=$INSTANCE_TYPE"
+    ARCH: x86_64
+    AMI_ID_ARG: "--x86-ami-id=$KERNEL_MATRIX_TESTING_X86_AMI_ID"
+    LibvirtSSHKey: $CI_PROJECT_DIR/libvirt_rsa-x86
+    TEST_COMPONENT: security-agent
+
+.upload_security_agent_tests:
+  stage: kernel_matrix_testing_prepare
+  allow_failure: true
+  rules: !reference [.on_security_agent_changes_or_manual]
+  before_script:
+    - !reference [.retrieve_linux_go_deps]
+    - !reference [.kernel_matrix_testing_new_profile]
+    - !reference [.write_ssh_key_file]
+  script:
+    # Build dependencies directory
+    - mkdir -p $DEPENDENCIES
+    - pushd $DEPENDENCIES
+    # copy clang and llc bins
+    - mkdir -p $EMBEDDED_BIN
+    - cp $CLANG_BPF $EMBEDDED_BIN
+    - cp $LLC_BPF $EMBEDDED_BIN
+    - mkdir -p $EMBEDDED_INC
+    # copy gotestsum and test2json
+    - mkdir -p $GO_BIN
+    - cp $GOTESTSUM $GO_BIN
+    - cp $TEST2JSON $GO_BIN
+    - mkdir junit
+    - mkdir testjson
+    - mkdir pkgjson
+    - popd
+    - pushd $CI_PROJECT_DIR/test/new-e2e/system-probe/test-runner && go build -o $DEPENDENCIES/test-runner && popd
+    - pushd $CI_PROJECT_DIR/test/new-e2e/system-probe/test-json-review && go build -o $DEPENDENCIES/test-json-review && popd
+    - cp $CI_PROJECT_DIR/test/new-e2e/system-probe/test-runner/files/*.json $DEPENDENCIES/
+    # package all the dependencies
+    - ls -la $DEPENDENCIES
+    - pushd $DD_AGENT_TESTING_DIR/$ARCH
+    - tar czvf ../$ARCHIVE_NAME dependencies
+    - popd
```

Comment:
It seems like a lot of this is the same as system-probe? Could we extract it?

Reply:
I've extracted the beginning, the rest is not exactly the same. I was planning on reviewing this after this PR and maybe more aggresive refactoring, possibly moving part of the code to invoke tasks to make it easier to test/develop and reutilize locally, but I wanted to get this working first and also talking with you and @usamasaqib to see if it was worth it and how to approach it.